components/ide/jetbrains/backend-plugin/src/main/kotlin/io/gitpod/jetbrains/remote/GitpodManager.kt

```diff
@@ -62,6 +62,7 @@ class GitpodManager : Disposable {
     }
 
     val devMode = System.getenv("JB_DEV").toBoolean()
+    var isAutoForwardingPorts = false
```

Comment:
I am not sure it is good idea, instead we should take the given local port and look up in JB prot manager whether there is a mapping for it and if so use actual local port

Reply:
I did it like this because we currently need to support JB IDEs v2022.1, which doesn't have the port forwarding feature. Doing so wouldn't need to move the files to 'stable' and 'latest' folders. That was the best I could think of to keep the changes minimal.
But let me review it.

Reply:
I see, so it is kind of a feature flag? I think you need an interface like:
```java
interface GitpodPortService {
   forward(localPort: int) Optional<int>
}
```

For stable you implement then `NullGitpodPortService` which always return null, so no tunnelling, for latest you are implementing the real one which is using JB Port Manager.